src/components/RenderRouter/ArchiveItem.tsx

```diff
@@ -0,0 +1,269 @@
+
+import React, { EventHandler, SyntheticEvent } from 'react';
+import { withRouter, RouteComponentProps } from 'react-router-dom';
+import "./ArchiveItem.scss"
+import Amplify from 'aws-amplify';
+import awsmobile from '../../aws-exports';
+import DataLoader, {
+    VideoByVideoTypeData,
+    SeriesByTypeData
+} from './DataLoader';
+import VideoOverlay from '../VideoOverlay/VideoOverlay';
+import format from 'date-fns/format';
+
+Amplify.configure(awsmobile);
+
+type listData = SeriesByTypeData | VideoByVideoTypeData
+
+interface Props extends RouteComponentProps {
+    content: any,
+    data: any,
+    pageConfig: any,
+}
+interface State {
+    listData: listData[]
+    overlayData: any
+    urlHistoryState: any
+    selectedYear: string
+    yearOptions: string[]
+    currentNextToken: string | null
+}
+class ArchiveItem extends React.Component<Props, State> {
+
+    constructor(props: Props) {
+        super(props);
+        this.state = {
+            listData: [],
+            overlayData: null,
+            urlHistoryState: window.history.state,
+            selectedYear: 'All',
+            yearOptions: ['All'],
+            currentNextToken: 'placeholder'
+        }
+    }
+
+    async componentDidMount() {
+        const dataLoaded = (data: listData[]) => this.setData(data);
+        const checkNext = (data: string | null) => this.setState({ currentNextToken: data })
+        const query = { selector: 'all', class: this.props.content.class, subclass: this.props.data, sortOrder: this.props.content.sortOrder };
+        switch (query.class) {
+            case 'series':
+                await DataLoader.getSeriesByType(query, dataLoaded, checkNext);
+                return;
+            case 'videos':
+                await DataLoader.getVideos(query, dataLoaded, checkNext);
+                return;
+            default:
+                console.error(`unknown list data type ${this.props.content.class}`);
+                return;
+        }
+    }
+
+    componentDidUpdate() {
+        console.log(this.state.currentNextToken)
+    }
+
+    binVideosByYear(data: VideoByVideoTypeData[]) {
+        const binned: VideoByVideoTypeData[][] = []
+        let temp = []
+        const years: string[] = ['All']
+        for (let i = 0; i < data.length; i++) {
+            temp.push(data[i])
+            if (data[i]?.publishedDate?.slice(0, 4) !== data[i + 1]?.publishedDate?.slice(0, 4)) {
+                const temp2 = data[i]?.publishedDate?.slice(0, 4)
+                if (temp2)
+                    years.push(temp2)
+                binned.push(temp)
+                temp = []
+            }
+        }
+
+        return { binned: binned, years: years }
+    }
+
+    binSeriesByYear(data: SeriesByTypeData[]) {
```

Comment:
You can combine this with the previous declaration by writing it as
```ts
binByYear<T>(data: T[], selector: (item: T) => string | undefined | null)
```
It also cleans up your implementation by making sure you don't call different methods on `i` and `i + 1`.